### DISCUSSION TITLE
Linking genes mutated with samples

### DISCUSSION THREAD

**Discussion by TheFukalite** [PoisonAlien/maftools#690] [Q&A]
I have been using maftools for the past two weeks for a school project, I was only wondering if there's a way to generate a file/table containing the genes mutated (Hugo_Symbol) in each of the samples (Tumor_Sample_Barcode)?

none of the summaries contains such information, although it is contained in two separate summaries (genesummary and samplesummary) but I'm not how to put the two together?

Thank you in advance!

**Comment by PoisonAlien** [comment]
Hi,
This is what `oncoplot` does visually. If you would like to generate such a matrix, check out the `mutCountMatrix` function. 
Or there is another internal function that can be accessed with `maftools:::createOncoMatrix()`. 
I hope this helps. Let me know if this is not what you expected.

**Reply by TheFukalite** [reply]
Thanks for your reply, this is very close to what I needed to achieve, basically what I'm trying to achieve is a file that contains Tumor Sample Barcodes matched with Hugo Symbols of genes mutated. 

`subsetMaf(maf = mymaf, genes = c('MSH2', 'MSH3'), mafObj = FALSE)`

Running this code above helped me get what I need, I'm just not sure how to export the data into some sort of table or csv because the output in rstudio looks sort of disorganized.

**Reply by PoisonAlien** [reply]
Try `write.mafSummary()`. It will write the summary of MAF object to output files.
Also, If you can provide an example for the exact output you're trying to generate, I can suggest more concrete solution.

**Reply by TheFukalite** [reply]
Thanks for your reply again! I have already did `write.mafSummary()` which outputs the genesummary and samplesummary but there's not way to link samples with the mutations found in them, what I am trying to achieve txt or csv showing (Gene mutated, variant classification and tumor sample barcode) within the same row.
I actually tried playing around and opened the maf file using excel, and sort of gotten what I needed where each mutation, variant classification, and tumor sample barcode were linked together, please see the photo attached for reference (Columns A, I, P) but opening the maf with excel is just not practical as there's a lot more in the maf than what I want to look for.

![Screenshot (6)](https://user-images.githubusercontent.com/81448608/112919630-62962780-90ff-11eb-91b9-24ef7e993643.png)

**Reply by PoisonAlien** [reply]
You could just write the columns that you are interested in, to a tsv file. 

e.g.,
```r
# replace `laml` with MAF object.
data.table::fwrite(x = laml@data[,.(Hugo_Symbol, Variant_Classification, Tumor_Sample_Barcode)], file = "my_data.tsv", sep = "\t")
```

**Reply by TheFukalite** [reply]
That works! Thank you so much!!
